cirq/circuits/circuit_test.py

```diff
@@ -631,6 +631,8 @@ def test_multiply(circuit_cls):
     assert c * 0 == circuit_cls()
     assert d * 0 == circuit_cls()
     assert d * 2 == circuit_cls([cirq.Moment([cirq.X(a)]), cirq.Moment([cirq.X(a)])])
+    assert d * np.array([2])[0] == circuit_cls([cirq.Moment([cirq.X(a)]), cirq.Moment([cirq.X(a)])])
```

Comment:
can you add some cases with explicit np.dtypes? E.g. I'm not sure we're covering np.int32, though you did add it to the implementation.

Reply:
+1